### ISSUE TITLE
[Bug]: Rate limited jobs with priority are not sorted correctly in getJobs

### ISSUE THREAD

**Issue by luddd3** [taskforcesh/bullmq#1929]
### Version

3.14.2

### Platform

NodeJS

### What happened?

After rate limiting jobs with priority, they are not fetched in fifo order anymore.

### How to reproduce.

```javascript
it('should get jobs in fifo order', async function () {
  this.timeout(6000);

  const numJobs = 4;
  const dynamicLimit = 250;
  const duration = 100;

  const worker = new Worker(
    queueName,
    async () => {
      await worker.rateLimit(dynamicLimit);
      throw Worker.RateLimitError();
    },
    {
      connection,
      limiter: {
        max: 1,
        duration,
      },
    },
  );

  for (let i = 1; i <= numJobs; i++) {
    await queue.add(`${i}`, {}, { priority: 10 });
  }

  await delay(dynamicLimit);

  const jobs = await queue.getJobs(['waiting'], 0, -1, true);
  expect(jobs.map(x => x.name)).to.eql(['1', '2', '3', '4']);

  await worker.close();
});
```

### Relevant log output

```shell
1) Rate Limiter
       when dynamic limit is used
         when priority is provided
           should get jobs in fifo order:

      AssertionError: expected [ '2', '3', '1', '4' ] to deeply equal [ '1', '2', '3', '4' ]
      + expected - actual

       [
      +  "1"
         "2"
         "3"
      -  "1"
         "4"
       ]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by manast** [comment]
Interesting, this was supposed to be fixed in the latest version...

**Comment by manast** [comment]
Ok, so after some discussion with @roggervalf I think this works as designed. Since the jobs have actually already been processed, then manually rate limited, it is not possible to guarantee the order within the same priority. If the jobs had different priorities then yes. But in this case, all jobs have the same priorities and they are going from wait to active, then back to wait. This case can never be guaranteed, even though it looks trivial for this particular example, imagine in general with concurrency workers and jobs being added continuously.

**Comment by luddd3** [comment]
I am using BullMq as the underlying system for a phone queue. The calls are picked from the queue by 1. priority and 2. time in queue. When a job (call) is processed I check if there are any phone agents available, if not I rate limit the queue for x seconds and then try again. In this case it is important for me that the queue follows the exact ordering.

Even if there are technical difficulties, I think it would be necessary for the queue to handle jobs in the expected order. As it is now, unless you have some alternative solution, I'm not able to use this library.

**Comment by manast** [comment]
I think that in your case it would be better to use manual job processing, so that you only try to consume new jobs if there are free agents:  https://docs.bullmq.io/patterns/manually-fetching-jobs

**Comment by roggervalf** [comment]
This should be fixed in versión 3.15.8

**Comment by luddd3** [comment]
Thanks for fixing this! :slightly_smiling_face:
